### PR TITLE
Avoid subtracting by sys.maxsize when something is bounded below by -sys.maxsize - 1

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #106716

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

### CODE REVIEW

File: torch/fx/experimental/symbolic_shapes.py [hunk 1]
```diff
@@ -2900,7 +2900,10 @@ def _maybe_evaluate_static(self, expr: "sympy.Expr", *, unbacked_only: bool = Fa
             # Don't do anything if we don't have a nontrivial lower bound
             # Also don't do anything if we asked only to simplify unbacked
             # SymInt
-            if vr.lower == -sympy.oo or (unbacked_only and k in self.var_to_val):
+            if (
+                vr.lower < (-sys.maxsize - 1) // 2 or
```
Comment (albanD) [inline torch/fx/experimental/symbolic_shapes.py:2904]: Why halfway through the negatives is your bound here?

Comment (ezyang) [reply torch/fx/experimental/symbolic_shapes.py:2904]: I mostly want to distinguish between "-sys.maxsize - 1" and "small negative numbers that aren't too negative". But I could probably also do vr.lower < 0 without too much trouble; happy to change it.

Comment (albanD) [reply torch/fx/experimental/symbolic_shapes.py:2904]: The halfway boundary seems a bit surprising, but if it is what you're after that sounds ok. I don't have a strong opinion.

**Review by albanD — APPROVED** [review]

**Pull request closed without merging**